### DISCUSSION TITLE
In Golang, how to set -local options for goimports?

### DISCUSSION THREAD

**Discussion by uynap** [neoclide/coc.nvim#4679] [Q&A]
In order to group the import packages, I want to do something like `goimports -local "my_loacal_pacakge"`. This is a feature supported by goimports. Here's an example https://github.com/golang/go/issues/12420. 
Both Goland and Vscode have ways to setup it. The setting is at https://stackoverflow.com/questions/38704769/golang-import-grouping-by-package and https://stackoverflow.com/questions/68765400/the-option-formatting-local-in-vscode-go-its-working.

For me, it's like passing an arguments to the gopls, so I found it at gopls documents. https://github.com/golang/tools/blob/master/gopls/doc/vim.md#cocnvim

I tried to create the coc-setting.vim below, but it doesn't work.
```
{
    "languageserver": {
        "go": {
            "command": "gopls",
            "rootPatterns": [
                "go.mod"
            ],
            "trace.server": "verbose",
            "filetypes": [
                "go"
            ],
            "initializationOptions": {
                "formatting.local": "github.com/my_loacal_pacakge"
            }
        }
    },
}
```
Can someone help me point what's the right settings in coc.nvim? TIA

**Comment by fannheyward** [comment]
Try 

```json
{
    "languageserver": {
        "go": {
            "command": "gopls",
            "rootPatterns": [
                "go.mod"
            ],
            "trace.server": "verbose",
            "filetypes": [
                "go"
            ],
            "initializationOptions": {
                "gopls": {
                     "formatting.local": "github.com/my_loacal_pacakge"
                }
            }
        }
    },
}
```

If you're using https://github.com/josa42/coc-go, you can use 

```
  "go.goplsOptions": {
    "local": "xxx",
  }
```

**Reply by uynap** [reply]
@fannheyward The first config block doesn't work. It's not a valid config.
Got `[coc.nvim] Invalid settings: unexpected gopls setting "gopls"`. Is there a place documented the config syntax for `initializationOptions`? It seems I just need a correct config.

I'm not using coc-go, but tried to install it. The `"go.goplsOptions": {"local": "xxx"}` works. 

btw. I'm using go1.20.3 on a mac with the latest `gopls` and `goimports`.

**Reply by uynap** [reply]
Ok. Just FYI. After installing `coc-go`, my previous config works. 
```
{
    "languageserver": {
        "go": {
            "command": "gopls",
            "rootPatterns": [
                "go.mod"
            ],
            "trace.server": "verbose",
            "filetypes": [
                "go"
            ],
            "initializationOptions": {
                "formatting.local": "github.com/my_loacal_pacakge"
            }
        }
    },
}
```

**Reply by fannheyward** [reply]
You don't need to set `languageserver` in coc-settings.json while using coc-go.

Try

```json
{
    "languageserver": {
        "go": {
            "command": "gopls",
            "rootPatterns": [
                "go.mod"
            ],
            "trace.server": "verbose",
            "filetypes": [
                "go"
            ],
            "initializationOptions": {
                "local": "github.com/my_loacal_pacakge"
            }
        }
    },
}
```
